aws/aws-parallelcluster
# [SubnetUpdate] Block subnet updates when using a managed Fsx for Lustre FileSystem

### Description of changes
- Cluster updates involving ComputeFleet SubnetIds when a managed Fsx for Lustre FileSystem is in use results in deleteion of the Fsx FileSystem.
- These changes introduce a new update policy for the SubnetIds which:
    - Checks if the cluster uses a managed Fsx for Lustre FileSystem
        - Shows an update validation error communicating the related risks and mitigation
    - Otherwise fall back to the QueueUpdateStrategy update policy

### Tests
- Unit tests:
    - If update includes SubnetIds and cluster configuration uses a managed Fsx for Lustre
    - Show validation failure message
    - If update includes SubnetIds and existing cluster configuration uses an External Fsx for Lustre FS
        - DO NOT show validation failure message
    - If update includes SubnetIds and new cluster configuration does NOT have a managed Fsx for Lustre FS
        - DO NOT show validation failure message
- Manual tests
    - Attempted running pcluster update after updating the ComputeFleet SubnetIds
    - Confirmed validation message is shown
    ```
    {
      "parameter": "Scheduling.SlurmQueues[queue1].Networking.SubnetIds",
      "requestedValue": [
        "subnet-0230367ab0e5123a4"
      ],
      "message": "Updating the compute fleet subnet will cause these FSx for Lustre file system(s) to be replaced: {'custom-fsx-lustre'}.. If you intend to proceed with the update, please make sure to back-up your data and explicitly replace the file system(s) ({'custom-fsx-lustre'}) with a new one(s) in the cluster configuration.",
      "currentValue": [
        "subnet-01b3415bfd0bbbee6"
      ]
    }
    ```

### References
* Link to related PRs in other packages (i.e. cookbook, node).
* Link to documentation useful to understand the changes.

### Checklist

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
